core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -49,7 +54,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     properties.remove(Properties.BatchReadableWritable.NAME);
     properties.remove(Properties.BatchReadableWritable.TYPE);
 
-    pipelineConfigurer.createDataset(datasetName, datasetType, DatasetProperties.builder().addAll(properties).build());
+    // null check for tests, as macro fields map and properties are null in test cases
+    if (pluginConfig.getProperties() != null && !pluginConfig.containsMacro(Properties.BatchReadableWritable.NAME)) {
```

Comment:
rather than doing this, I think it would be better to have a BatchWritableConfig with a single macro enabled name field that all the other configs extend.